src/app/3d/qgs3dnavigationwidget.cpp

```diff
@@ -207,36 +199,36 @@ Qgs3DNavigationWidget::Qgs3DNavigationWidget( Qgs3DMapCanvas *parent ) : QWidget
   mCameraInfo->horizontalHeader()->hide();
   mCameraInfo->horizontalHeader()->setSectionResizeMode( QHeaderView::ResizeMode::Stretch );
 
-  QGridLayout *gridLayout = new QGridLayout( this );
-  gridLayout->addWidget( mTiltUpButton, 0, 0 );
-  gridLayout->addWidget( mTiltDownButton, 3, 0 );
-  gridLayout->addWidget( mZoomInButton, 0, 3 );
-  gridLayout->addWidget( mZoomOutButton, 3, 3 );
-  gridLayout->addWidget( mCompas, 1, 1, 2, 2 );
-  gridLayout->addWidget( mMoveUpButton, 0, 1, 1, 2, Qt::AlignCenter );
-  gridLayout->addWidget( mMoveRightButton, 1, 3, 2, 1, Qt::AlignCenter );
-  gridLayout->addWidget( mMoveDownButton, 3, 1, 1, 2, Qt::AlignCenter );
-  gridLayout->addWidget( mMoveLeftButton, 1, 0, 2, 1, Qt::AlignCenter );
-
-  QHBoxLayout *layout = new QHBoxLayout;
-  layout->addWidget( mCameraInfo );
   mCameraInfo->setVisible( false );
+  QObject::connect( mCameraInfoCheckBox, &QCheckBox::clicked, parent, [ = ]( bool enabled ) { mCameraInfo->setVisible( enabled ); } );
 
-  QCheckBox *cameraInfoCheckBox = new QCheckBox( this );
-  cameraInfoCheckBox->setText( tr( "Show camera info (for debugging)" ) );
-  cameraInfoCheckBox->setChecked( false );
-  QObject::connect( cameraInfoCheckBox, &QCheckBox::clicked, parent, [ = ]( bool enabled ) { mCameraInfo->setVisible( enabled ); } );
+  mWidgetLayout->setAlignment( Qt::AlignTop );
 
-  gridLayout->addWidget( cameraInfoCheckBox, 4, 0, 1, 4, Qt::AlignLeft );
-  gridLayout->addLayout( layout, 5, 0, 1, 4, Qt::AlignCenter );
+  QObject::connect( mCbo3DAxis, &QComboBox::currentTextChanged, this, &Qgs3DNavigationWidget::updateAxisMode );
+  mCbo3DAxis->setCurrentIndex( 1 ); // not OFF
+}
 
-  gridLayout->setAlignment( Qt::AlignTop );
+void Qgs3DNavigationWidget::updateAxisMode( const QString &modeStr )
+{
+  if ( mParent3DMapCanvas->get3DAxis() )
+  {
+    Qgs3DAxis::Mode m;
+    if ( modeStr.toUpper() == "SRS" )
```

Comment:
string comparison is fragile here, as it will break with translations. Better to use the user data approach we use elsewhere (eg in https://github.com/qgis/QGIS/commit/a927a761feaecb2304f4b42410a61ecba1a9a5f6#diff-91e6a9080a8dbf15564686a5e9ba262f4b36b920eb8093a124aed16ba9f2e49eR147, https://github.com/qgis/QGIS/commit/a927a761feaecb2304f4b42410a61ecba1a9a5f6#diff-91e6a9080a8dbf15564686a5e9ba262f4b36b920eb8093a124aed16ba9f2e49eR147 )